### PR TITLE
Fix crash when multiple share dialogs were shown

### PR DESCRIPTION
**Pull request by erikjv**
Fixes: #11673

### CODE REVIEW

File: src/gui/owncloudgui.cpp [hunk 1]
```diff
@@ -1063,35 +1063,21 @@ void ownCloudGui::slotShowShareDialog(const QString &sharePath, const QString &l
             maxSharingPermissions = SharePermission(0);
         }
 
-
-        ShareDialog *w = nullptr;
-        if (_shareDialogs.contains(localPath) && _shareDialogs[localPath]) {
-            qCInfo(lcApplication) << "Raising share dialog" << sharePath << localPath;
-            w = _shareDialogs[localPath];
+        if (_shareDialog && _shareDialog->localPath() == localPath) {
```
Comment (TheOneRing) [inline src/gui/owncloudgui.cpp:1066]: I guess we should return in that case, not sure what happens when addModalLegacyDialog is called multiple times with the same widget.

File: src/gui/owncloudgui.cpp [hunk 2]
```diff
@@ -1063,35 +1063,21 @@ void ownCloudGui::slotShowShareDialog(const QString &sharePath, const QString &l
...
         } else {
-            qCInfo(lcApplication) << "Opening share dialog" << sharePath << localPath << maxSharingPermissions;
-            w = new ShareDialog(accountState, folder->webDavUrl(), sharePath, localPath, maxSharingPermissions, startPage, settingsDialog());
-            w->setAttribute(Qt::WA_DeleteOnClose, true);
-
-            _shareDialogs[localPath] = w;
-            connect(w, &QObject::destroyed, this, &ownCloudGui::slotRemoveDestroyedShareDialogs);
+            qCInfo(lcApplication) << "Opening new share dialog" << sharePath << localPath << maxSharingPermissions;
+            if (_shareDialog) {
+                _shareDialog->close();
```
Comment (TheOneRing) [inline src/gui/owncloudgui.cpp:1071]: It should also work if we stack them, so should add another one?

Comment (erikjv) [reply src/gui/owncloudgui.cpp:1073]: No, just 1 sharing dialog for the latest. If a user requested one before but never used it to copy the link, then 🤷, apparently it wasn't important.

**Comment by erikjv** [comment]
Note: document that this is oC10 only functionality.

**Review by TheOneRing — APPROVED** [review]

**Pull request merged**